src/main/target/NOX/target.c

```diff
@@ -31,7 +31,12 @@ const timerHardware_t timerHardware[USABLE_TIMER_CHANNEL_COUNT] = {
     DEF_TIM(TIM2,  CH3,  PB10, TIM_USE_PPM,   0,     0), // T2C3
 
     DEF_TIM(TIM5,  CH2,  PA1,  TIM_USE_MOTOR, 0, 0), // T2C2(1,6,3), T5C2(1,4,6)
+    // Use T3C2 instead of T1C1N to enable DSHOT telemetry
+#ifdef USE_DSHOT_TELEMETRY
```

Comment:
Is there a reason to not permanently change this?

Reply:
You tell me 😄. This feature is the first time I work with the low level timers, dma etc for this chip. I don't know that I can fully guarantee that there will be no negative side effects from making this change.

Reply:
Maybe @jflyper can comment?

Reply:
Well, TIM3CH2 either works for this pin, or it doesn't. If it does then it there is no reason to not always use it, if it doesn't then it should not be used at all. Introducing an either / or allocation will just make it harder to make sense of problem reports.

Reply:
https://github.com/betaflight/betaflight/blob/131e32c142cf4da3ffddad0078d32579eda7c5ce/src/main/drivers/pwm_output.h#L44

Sorry, my bad, it's called `MOTOR_STOP`, but anyhow, Dshot commands will only be accepted if this (and not a motor speed value) has been sent as the last speed setting.

In any case, even waiting 5 seconds will not necessarily guarantee that the ESCs are ready (think when the flight controller is powered from USB while bench testing, and a battery is plugged in at a later stage).

Your best bet is probably to add the commands to `tryArm()` (https://github.com/betaflight/betaflight/blob/131e32c142cf4da3ffddad0078d32579eda7c5ce/src/main/fc/core.c#L367), just like what is done for reversing the motor directions for crash flip. This will also ensure that the motors are stopped / disarmed at the time.

Reply:
It certainly works for this pin - that's not the issue I see. But different timers mean different dma channels which could lead to conflicts. That's my worry since I'm not sure I know all uses of dma in BF.

If there are no risks I agree switching over completely makes more sense.

Reply:
If we use TIM3_CH2 instead of TIM1_CH1N for motor 2, new DMA spec DMA1_Stream5 collides with LED strip (TIM2_CH1, DMA1_Stream5).

Following is DMA stream allocation for stock NOX1 with Dshot, LED strip and ADC turned on.
```
# dma

Currently active DMA:
--------------------
DMA1 Stream 0: FREE
DMA1 Stream 1: FREE
DMA1 Stream 2: MOTOR 4
DMA1 Stream 3: FREE
DMA1 Stream 4: MOTOR 1
DMA1 Stream 5: LED_STRIP
DMA1 Stream 6: FREE
DMA1 Stream 7: MOTOR 3
DMA2 Stream 0: FREE
DMA2 Stream 1: FREE
DMA2 Stream 2: FREE
DMA2 Stream 3: FREE
DMA2 Stream 4: ADC
DMA2 Stream 5: FREE
DMA2 Stream 6: MOTOR 2
DMA2 Stream 7: FREE
```

Reply:
Alright. I'd say we keep it conditional for now then but keep this in mind to be addressed with the potential dshot driver rework using gpios as peripherals instead of timer pins.